### ISSUE TITLE
Require reports go through !info for submissions and comments.

### ISSUE THREAD

**Issue by Tenpaii** [Dougley/MBv2#111]
**Is your feature request related to a problem? Please describe.**
When people block you, you obviously can't react to their post thus hindering you from reporting whatever link they're trying to report.

**Describe the solution you'd like**
Only have :report: pop-up when !info is used.

Also, would it be possible to show the content of a comment through !info as well?

**Additional context**

**Comment by blastoise186** [comment]
Personally, I'd rather have it so that reports don't have to go via the into command only, there's a few places where it's not appropriate because the post would explode a channel.

However, the info command supporting comments would be useful, especially if it can be done in the same way as posts

**Comment by Tenpaii** [comment]
> there's a few places where it's not appropriate because the post would explode a channel.

Please do cite example/s of this.

**Comment by Dougley** [comment]
I don't see an immediate need to implement this, however if you have some arguments for this please reply.

**Comment by Zacatero** [comment]
@Dougley a good example was the other day, Ghoul posted like 10 in a row but she just used the link. Ghoul has blocked me so I couldn't react to any of those. I agree that the reactions should only be added if its a !info or something that the bot posts, not when people post it.

**Comment by blastoise186** [comment]
The problem with forcing this being command only is that it makes it harder to get reports done via Mobile, especially when the keyboard has things like "Auto-insert Space after punctuation" turned on.

However, I would like the ability for the info command to support comments, because it would make it easier to know which comment is being reported. I'm just wondering whether it might be a good idea to break this item out into its own Issue however

**Comment by Zacatero** [comment]
@blastoise186 But !info works with the link doesnt it? There are several hard things to do on mobile, including dupes, don't get me wrong. BUT doing !Info is pretty simple.

**Comment by blastoise186** [comment]
@Zacatero the info command doesn't currently work with comments, Ghoul agreed to post the comment links on my behalf as a workaround for now. However, it's painful to have to remember to backspace after ! on mobile to avoid getting ! info and disabling the auto-space option on mobile keyboards gets annoying for other things.

I feel like comments are a different feature request though, so I'll look at breaking that out

**Comment by Tenpaii** [comment]
@blastoise186 This honestly seems like a "you" problem instead of widespread/majority with the whole "auto-insert space after punctuation" setting. Like how many of the Custodians has this setting on that it'd be a problem for the majority? Imo, one backspace isn't as big of an issue than not being able to report.

@Dougley Someone has a quite a few people blocked that they needed to ask Ghoul to post the links so we can react to it. I get that people are free to block whoever they want for whatever reason but it puts the blocked person on an unfair level. At least with this, it levels the playing field since the bot can't block anyone and it would mean that everyone can block whoever they want without the other getting affected in regards to Custodian duties.

**Comment by Zacatero** [comment]
I mean, in all honesty, being blocked shouldn't prevent you from adding reactions to ones that already exists.... It really should just limit you to adding new reactions. Like the permission. If that were fixed this really wouldn't be an issue.

**Comment by Tenpaii** [comment]
I agree however since that's an issue with Discord itself, this is a workaround until it changes or IF it even changes in the first place.

**Comment by MoonlightCapital** [comment]
>Ghoul has blocked me

>Ghoul agreed to post the comment links on my behalf

@Zacatero @blastoise186 those statements contradict each other, please explain the reason why Ghoul would post stuff in your behalf if she has blocked people.

Now, not to get on Ghoul's business or anything, but reporting shouldn't be a "ok i'll unblock you" free card.

**Comment by Tenpaii** [comment]
@MoonlightCapital Ghoul has Zac blocked while Blastoise asked Ghoul to post the links in chat because he had a few people blocked.

No one's asking anyone to be unblocked or no one's saying they should be unblocked just to report things. The suggestion is literally so people don't need to care about blocking.

**Comment by Dougley** [comment]
The conversation upto this point seems to detract from the original point of the issue, please stay ontopic

**Comment by SuperSajuuk** [comment]
I agree with @Tenpaii. All reports and votes should be done through the bot to keep the entire system fair and free of any impartiality, thus removing any care for people blocking each other or users they don't like.

@Dougley Actually, all the conversation here is in relation to the suggestion and in fact, the particular case of a specific user (and since it's been mentioned anyway, it's Blastoise) blocking a large number of active members for seemingly no reason (in some cases) and then posting links to trigger the reactions without using the bot to pull up the information, is what triggered this thread to be created.

By doing what said user actually did as described earlier, the user in question has, in effect, manipulated the reporting system to make sure only the people he didn't have blocked were able to report when he posted links in the chat, thus in effect blocking numerous users from getting any XP from the things he wanted to be reported.

Since bots have no way of blocking users from reactions (except by having a hand coded system to do this manually, rather than it being an API limitation), it makes it fair on everyone by having all actions go through the bot. If someone is blocked by the bot, then it would have been a moderator doing so and the user would have been told by said mod that they're prevented from using the bot reactions. Just like how you can block users from the leaderboard or regaining custodian.

**Comment by Zacatero** [comment]
While I agree this is something that should be added, I also acknowledge that if Discord would just make the block reactions act like disallowing the reaction permission (meaning you cant add new ones, but you can contribute to ones that others have put) then this wouldnt be an issue, [but people didn't really seem to like that idea](https://support.discordapp.com/hc/en-us/community/posts/360043267811)

**Comment by Dougley** [comment]
I'm going to start a pilot with this change momentarily, since I agree that the way discord handles reactions for blocked users is extremely disadvantageous to the system.

**Comment by Dougley** [comment]
This change is considered successful
